### PR TITLE
fixed recovery of axes_block from structure describing 0-dim dataset

### PR DESCRIPTION
**Pull request by abuts**
unit test and missing logic which allow to read old 0-dimensional data. 
The problem was in recovering `axes_block` from old 0-dimensional data and the problem is fixed.

### CODE REVIEW

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m [hunk 1]
```diff
@@ -27,3 +27,8 @@
     in.label = in.ulabel;
     in = rmfield(in,'ulabel');
 end
+if sum(nbins_all_dims) == 4
```
Comment (oerc0122) [inline horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m:30]: Slightly cryptic way of saying 
```
if all(nbins_all_dims == 1)
```

File: horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m [hunk 2]
```diff
@@ -27,3 +27,8 @@
...
+    if isfield(in,'dax')
```
Comment (oerc0122) [inline horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m:31]: Should be merged with previous `if` and `&&`

**Review by oerc0122 — CHANGES_REQUESTED** [review]

File: _test/test_transformation/test_AxesBlockBase_properties.m [hunk 3]
```diff
@@ -146,6 +146,32 @@ function test_axes_block_nodes_hull(~)
...
+                'iax', zeros(1,0),...
+                'iint',zeros(2,0),...
+                'pax', [1 2 3 4],...
+                'p', [],...
+                'dax', [1 2 3 4],...
+                's', 65424884,...
+                'e', 560235008,...
+                'npix', 5028200);
+            old_data.label = {'Q_\zeta'  'Q_\xi'  'Q_\eta'  'E'};
+            old_data.p = {[-1;1], [-1;1]  [-1;1] [-1;1]};
```
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_transformation/test_AxesBlockBase_properties.m:170]: Due to Matlab rules if you give the structure cellarray input, the result would be array of structures.  To get structure requested here you would need to do what I did.

[the remark this answers is not in the document]
Comment (oerc0122) [reply _test/test_transformation/test_AxesBlockBase_properties.m:170]: You need to wrap the cellarray in another cell array, i.e.:
````
'label', {{'Q_\zeta'  'Q_\xi'  'Q_\eta'  'E'}}
```

File: horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m (hunk 1 above)
Comment (abuts) [reply horace_core/sqw/axes_blocks/@line_axes/private/convert_old_struct_into_nbins_.m:30]: well, probably its alternative way of doing this.

File: _test/test_transformation/test_AxesBlockBase_properties.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_transformation/test_AxesBlockBase_properties.m:170]: or this

**Pull request merged**